### ISSUE TITLE
IPv6 Tip

### ISSUE THREAD

**Issue by eurobank** [CIDRAM/CIDRAM#227]
Today i was checking the statistics and saw zero IPv6 blocks etc. So i asked my hosting about it and i was told that IPv6 is not provided to their hosting service. I asked around and found that this is the case to most such services.

Is that correct?

Anyways, i disabled the IPv6 rules and i think i gained a bit of performance. 

Just a tip for other users. 

If i'm not correct, please step in ....

**Comment by 737simpilot** [comment]
That sounds like a crap host. I've been with three hosts and all had IPv6 ability. That's really ridiculous that you can't receive IPv6 traffic. I get just as much IPv6 traffic as IPv4 traffic so you're being limited on who can connect to your site.

**Comment by eurobank** [comment]
> 
> 
> That sounds like a crap host. I've been with three hosts and all had IPv6 ability. That's really ridiculous that you can't receive IPv6 traffic. I get just as much IPv6 traffic as IPv4 traffic so you're being limited on who can connect to your site.

Not a crap host, i assure you. I do not want to post company names here.

Still, it puzzles me that no IPv6 traffic, maybe i have to research more this.

**Comment by eurobank** [comment]
Do you all have any IPv6 Block numbers in your Statistics? or it is just me? Can you check?

![Screenshot](https://user-images.githubusercontent.com/4467158/119115129-8014ad00-ba2f-11eb-8d1c-392fe80ebe3d.png)

**Comment by eurobank** [comment]
Ok, i do not have web IP6 connectivity in my hosting plan. I have checked a lot of sites and most do not have that also. Some do. Most do not.

So if you DO NOT have IPv6, then disabling the IPv6 signatures is a good idea.

**Comment by Maikuolan** [comment]
I don't run too many websites of my own anymore (between work, hobbies, domestic responsibilities, etc, there's just not enough time or incentive for it really, and for the most part, I get enough enjoyment and satisfaction from just maintaining code and open-source software like CIDRAM et al). Though, of those few that I do, about half accept IPv6 connectivity, and about half don't.

IPv6 is just another one of those things, I think, which most webhosting providers *should've* already adopted by now, but in reality, in many cases haven't, for whatever reasons/excuses/etc is the case (e.g., don't need it, too expensive, too complicated, etc). Though, in time, the ratio of those that do versus those that don't will definitely be eventually shifting towards those that do, due to more of the world gradually becoming connected to the internet, the ever-growing number of internet-ready devices the average person uses every day nowadays, the gradual depletion of available IPv4 address space, etc and etc. There'll eventually come a time, too, when those hosts which don't provide IPv6, will effectively not be able to connect to the majority of the wider internet anymore. But, that time isn't going to be this year, next year, or any time too soon into the immediate future, and also isn't something really within the scope of this discussion here anyhow.

TL;DR: If you don't need some of the signature files, better to be without them, I think. So yeah, this (generally speaking) makes sense.

> Anyways, i disabled the IPv6 rules and i think i gained a bit of performance.

Seeing as signature files should only be loaded by CIDRAM when they're actually needed anyway (e.g., IPv4 signature files for IPv4 requests, IPv6 signature files for IPv6 requests, etc), removing unnecessary signature files shouldn't, in theory, really make much difference to performance. CIDRAM tries to determine the type of request it's looking at before loading any signature files, and only tries to load the signature files corresponding to the type of request it's looking at.

However, that all said.. I wouldn't discourage disabling or getting rid of any unnecessary files anyway, seeing as such unnecessary files are serving only to use up available disk space which could be used elsewhere, or just left unused (plus, keeping them would mean more files to regularly update, which would mean more bandwidth used at least during the update process, would mean more work to do when restoring or backing up your website and all its files, etc). It's easy enough to simply reinstall and reactivate them again at a later date anyway, if you discover at such a time that you're needing them again.

**Comment by eurobank** [comment]
> However, that all said.. I wouldn't discourage disabling or getting rid of any unnecessary files anyway, seeing as such unnecessary files are serving only to use up available disk space which could be used elsewhere, or just left unused (plus, keeping them would mean more files to regularly update, which would mean more bandwidth used at least during the update process, would mean more work to do when restoring or backing up your website and all its files, etc). It's easy enough to simply reinstall and reactivate them again at a later date anyway, if you discover at such a time that you're needing them again.

It is not a matter of space. I _think_ i see a little more energy in the sites in my shared hosting plan. Hard to tell or measure when on shared, maybe just my idea. Cpanel stats take a while to reflect all that. I will check after a week and compare cpu/disk/memory and if i find something interesting, i will update here.

**Comment by Maikuolan** [comment]
> I will check after a week and compare cpu/disk/memory and if i find something interesting, i will update here.

Sounds good, and looking forward to seeing the results. :-)

**Comment by 737simpilot** [comment]
If you have a PHP file that's prepended via the htaccess file to launch CIDRAM, then use this code in that PHP file. You'll have to tailor to your environment. 


`<?php
$time_start = microtime(true);
require '/home/user_name/public_html/cidram/loader.php';
$time_end = microtime(true);
$execution_time = ($time_end - $time_start);
echo '<b>CIDRAM Total Execution Time:</b> '.$execution_time.' Secs'`
